src/main/java/software/amazon/awssdk/crt/mqtt5/Mqtt5ListenerOptions.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.mqtt5;
+
+import software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions.LifecycleEvents;
+
+/**
+ * Configuration for the creation of Mqtt5Listener
+ *
+ * MQTT5 support is currently in <b>developer preview</b>.  We encourage feedback at all times, but feedback during the
+ * preview window is especially valuable in shaping the final product.  During the preview period we may make
+ * backwards-incompatible changes to the public API, but in general, this is something we will try our best to avoid.
+ */
+public class Mqtt5ListenerOptions {
```

Comment:
Just a quick follow-up: I looked at the design document and it looks like the PR is currently missing the Mqtt5ListenerOptionsBuilder class, and that's how the events will be set.

Reply:
I was a little hesitate if the builder is necessary as ideally the Listener will be only used internally by us. And obviously I forget about the builder afterwards:(. Regardless, I figured it would be easier to remove it in the future. Thanks for reminder!